src/main/webapp/WEB-INF/view/chat.jsp

```diff
@@ -21,78 +22,59 @@
 Conversation conversation = (Conversation) request.getAttribute("conversation");
 List<Message> messages = (List<Message>) request.getAttribute("messages");
 %>
+<t:base>
+    <jsp:attribute name="title"><%= conversation.getTitle() %></jsp:attribute>
+    <jsp:attribute name="headContent">
+         <style>
+            #chat {
+              background-color: white;
+              height: 500px;
+              overflow-y: scroll
+            }
+          </style>
+
+          <script>
+            // scroll the chat div to the bottom
+            function scrollChat() {
+              var chatDiv = document.getElementById('chat');
+              chatDiv.scrollTop = chatDiv.scrollHeight;
+            };
+            window.onload = scrollChat
+          </script>
+    </jsp:attribute>
+    <jsp:attribute name="bodyContent">
+            <h1><%= conversation.getTitle() %>
+              <a href="" style="float: right">&#8635;</a></h1>
+
+            <hr/>
+
+            <div id="chat">
+              <ul>
+            <%
+              for (Message message : messages) {
+                String author = UserStore.getInstance()
+                  .getUser(message.getAuthorId()).getName();
+            %>
+              <li><strong><%= author %>:</strong> <%= message.getContent() %></li>
+            <%
+              }
+            %>
+              </ul>
+            </div>
+
+            <hr/>
+
+            <% if (request.getSession().getAttribute("user") != null) { %>
+            <form action="/chat/<%= conversation.getTitle() %>" method="POST">
+                <input type="text" name="message">
+                <br/>
+                <button type="submit">Send</button>
+            </form>
+            <% } else { %>
+              <p><a href="/login">Login</a> to send a message.</p>
+            <% } %>
+
+            <hr/>
+    </jsp:attribute>
+</t:base>
 
-<!DOCTYPE html>
-<html>
-<head>
-  <title><%= conversation.getTitle() %></title>
-  <link rel="stylesheet" href="/css/main.css" type="text/css">
-
-  <style>
-    #chat {
-      background-color: white;
-      height: 500px;
-      overflow-y: scroll
-    }
-  </style>
-
-  <script>
-    // scroll the chat div to the bottom
-    function scrollChat() {
-      var chatDiv = document.getElementById('chat');
-      chatDiv.scrollTop = chatDiv.scrollHeight;
-    };
-  </script>
```

Comment:
The `style` and `script` tags have been moved to `<jsp:attribute name="headContent">`